packages/core/src/serialization/EntitySerializer.ts

```diff
@@ -229,7 +229,7 @@ export class EntitySerializer {
       return this.serialize(child, childOptions) as EntityValue<T>;
     }
 
-    const pk = wrapped.getPrimaryKey(true)!;
+    const pk = wrapped.getPrimaryKey(false)!;
```

Comment:
the very same issue will be in implicit serialization, lets alter the test to try that too and fix that too, its in `EntityTransformer` class

```suggestion
    const pk = wrapped.getPrimaryKey()!;
```